anndata/_core/anndata.py

```diff
@@ -683,6 +684,10 @@ def X(self, value: Optional[Union[np.ndarray, sparse.spmatrix]]):
                 f"need to be {self.shape}."
             )
 
+    @X.deleter
+    def X(self):
+        self.X = None
```

Comment:
With other attributes, deleting them when an object is a view causes it to become an "actual" object. This happens for any modification. For example:

```python
In [1]: import scanpy as sc
pbmc
In [2]: pbmc = sc.datasets.pbmc3k_processed()

In [3]: v = pbmc[:, :50]

In [4]: v
Out[4]: 
View of AnnData object with n_obs × n_vars = 2638 × 50
    obs: 'n_genes', 'percent_mito', 'n_counts', 'louvain'
    var: 'n_cells'
    uns: 'draw_graph', 'louvain', 'louvain_colors', 'neighbors', 'pca', 'rank_genes_groups'
    obsm: 'X_pca', 'X_tsne', 'X_umap', 'X_draw_graph_fr'
    varm: 'PCs'
    obsp: 'distances', 'connectivities'

In [5]: del v.obsm

In [6]: v
Out[6]: 
AnnData object with n_obs × n_vars = 2638 × 50
    obs: 'n_genes', 'percent_mito', 'n_counts', 'louvain'
    var: 'n_cells'
    uns: 'draw_graph', 'louvain', 'louvain_colors', 'neighbors', 'pca', 'rank_genes_groups'
    varm: 'PCs'
    obsp: 'distances', 'connectivities'
```

That should happen here too. Could you add a test similar to `test_attr_deletion` which checks for this?

I think it's fair to say that `adata.X = None` should be equivalent to `del adata.X`, so you could also make sure that works when the `AnnData` is a view? It looks like it's set to throw an error right now.

Reply:
Oh, the lines where I should have commented this are 628-631, where:

```python
        if value is None:
             if self.is_view:
                 raise ValueError(
                    "Copy the view before setting the data matrix to `None`."
                )
```

Reply:
See 72618c3 -- is this what you meant?